eclair-core/src/test/scala/fr/acinq/eclair/payment/PostRestartHtlcCleanerSpec.scala

```diff
@@ -543,6 +544,99 @@ class PostRestartHtlcCleanerSpec extends TestKitBaseClass with FixtureAnyFunSuit
+      def channelId: ByteVector32 = channelId_ab_1
+      def announceChannel: Boolean = false
+    }
+
+    // Non-standard channel goes to NORMAL state:
+    system.eventStream.publish(ChannelStateChanged(channel.ref, channelId_ab_1, system.deadLetters, a, OFFLINE, NORMAL, Some(cs)))
+    channel.expectMsg(CMD_FAIL_HTLC(1L, Right(TemporaryNodeFailure), commit = true))
+    channel.expectNoMsg(100 millis)
+  }
+
+  test("Relayed standard->nonstandard HTLC is retained") { f =>
+    import f._
+
+    val relayedPaymentHash = randomBytes32
+    val trampolineRelayedPaymentHash = randomBytes32
+    val trampolineRelayed = Origin.TrampolineRelayedCold((channelId_ab_2, 0L) :: Nil)
+    val relayedHtlcIn = buildHtlcIn(0L, channelId_ab_2, trampolineRelayedPaymentHash)
+    val nonRelayedHtlcIn = buildHtlcIn(1L, channelId_ab_2, relayedPaymentHash)
+
+    val pluginParams = new CustomCommitmentsPlugin {
+      def name = "test with outgoing HTLC to remote"
+      def getIncomingHtlcs: Seq[PostRestartHtlcCleaner.IncomingHtlc] = List.empty
+      def getHtlcsRelayedOut(htlcsIn: Seq[PostRestartHtlcCleaner.IncomingHtlc]): Map[Origin, Set[(ByteVector32, Long)]] = Map(trampolineRelayed -> Set((channelId_ab_2, 0L)))
+    }
+
+    val nodeParams1 = nodeParams.copy(pluginParams = List(pluginParams))
+    val c = ChannelCodecsSpec.makeChannelDataNormal(List(relayedHtlcIn, nonRelayedHtlcIn), Map.empty)
+    nodeParams1.db.channels.addOrUpdateChannel(c)
+
+    val channel = TestProbe()
+    f.createRelayer(nodeParams1)
+    register.expectNoMsg(100 millis) // nothing should happen while channels are still offline.
+
+    // Standard channel goes to NORMAL state:
+    system.eventStream.publish(ChannelStateChanged(channel.ref, c.commitments.channelId, system.deadLetters, a, OFFLINE, NORMAL, Some(c.commitments)))
+    channel.expectMsg(CMD_FAIL_HTLC(1L, Right(TemporaryNodeFailure), commit = true))
+    channel.expectNoMsg(100 millis)
+  }
+
+  test("Non-standard HTLC CMD_FAIL in relayDb is retained") { f =>
+    import f._
+
+    val trampolineRelayedPaymentHash = randomBytes32
+    val relayedHtlc1In = buildHtlcIn(0L, channelId_ab_1, trampolineRelayedPaymentHash)
+
+    val pluginParams = new CustomCommitmentsPlugin {
+      def name = "test with incoming HTLC from remote"
+      def getIncomingHtlcs: Seq[PostRestartHtlcCleaner.IncomingHtlc] = List(PostRestartHtlcCleaner.IncomingHtlc(relayedHtlc1In.add, None))
+      def getHtlcsRelayedOut(htlcsIn: Seq[PostRestartHtlcCleaner.IncomingHtlc]): Map[Origin, Set[(ByteVector32, Long)]] = Map.empty
+    }
+
+    val cmd = CMD_FAIL_HTLC(id = 0L, reason = Left(ByteVector.empty), replyTo_opt = None)
+    val nodeParams1 = nodeParams.copy(pluginParams = List(pluginParams))
+    nodeParams1.db.pendingRelay.addPendingRelay(channelId_ab_1, cmd)
+    f.createRelayer(nodeParams1)
+    register.expectNoMsg(100 millis)
+    assert(nodeParams1.db.pendingRelay.listPendingRelay(channelId_ab_1).head == cmd)
```

Comment:
You need an `awaitCond` otherwise there's a race condition (you might hit your assert before the PostRestartHtlcCleaner has the opportunity to call `cleanUpRelayDb`).